Added FollowPathBehaviour, TeleportBehavior, and a Trail

steffanmouton opened the pull request.

steffanmouton committed:
Added FollowPathBehaviour, TeleportBehavior, and a Trail

steffanmouton committed:
Merge branch 'master' into steffan

chuwilliamson commented on Assets/Scripts/Steffan/TeleportBehaviour.cs:
```diff
@@ -0,0 +1,35 @@
+
+	public UnityEvent response;
+	// Use this for initialization
+	void Start () {
```
remove unused functions

chuwilliamson commented on Assets/Scripts/Steffan/TeleportBehaviour.cs:
```diff
@@ -0,0 +1,35 @@
+	}
+	
+	// Update is called once per frame
+	void Update () {
```
move this into an InputBehaviour class that will be given the references

chuwilliamson commented on Assets/Scripts/Steffan/FollowPathBehaviour.cs:
```diff
@@ -0,0 +1,107 @@
+	void RestartPath()
+	{
+		WaypointsToFollow = new List<GameObject>(SavedPath);
+		transform.position = WaypointsToFollow[0].transform.position;
```
use a list of transforms in place of a list of gameObjects

chuwilliamson commented on Assets/Scripts/Steffan/FollowPathBehaviour.cs:
```diff
@@ -0,0 +1,107 @@
+	
+	private void OnTriggerEnter(Collider other)
+	{
+		if (!other.gameObject.CompareTag("Waypoint"))
```
make a string field with the Cinemachine.TagField attribute

steffanmouton replied:
Idk what this does, but yessir.

chuwilliamson reviewed (changes requested):
Changes should result in an approved request

steffanmouton committed:
@chuwilliamson pull request comments addressed

Added CinemachineTagField
Changed List<GameObject> to List<Transform>
Removed unused functions

steffanmouton commented:
Requested Changes addressed
Added Cinemachine to project
